script.js

```diff
@@ -1,4 +1,205 @@
+// Global variables
+var playerWins = 0;
+var playerLosses = 0;
+var playerDraws = 0;
+var gameState = "username";
+var userName = "";
+var gameReverse = 0;
+var lastWinner = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Game state: Confirm username
+  if (gameState == "username") {
+    userName = input;
+    gameState = "normal";
+    return `Hello ${userName}! Scissors, paper or stone?`;
+  }
+
+  // Game state changing
+  if (input == "reverse") {
+    gameReverse = 1;
+    return `Time to play in reverse~!`;
+  }
+
+  if (input == "korean") {
+    gameState = "korean";
+    return `Korean mode enabled!`;
+  }
+
+  if (input == "normal") {
+    gameState = "normal";
+    return `Normal mode enabled!`;
+  }
+
+  if (input == "computer") {
+    gameState = "computer";
+    return `Computer mode enabled!`;
+  }
+
+  // Game state: Main game
+  if (inputValidator(input)) {
+    return "Please input scissors, paper or stone.";
+  }
+  var checkReverse = gameReverse + determineReverse(input);
+
+  var playerRPS = input;
+  var botRPS = giveRandomRPS();
+  var verifyWinner = "";
+  var koreanWinner = "";
+
+  // Check computer mode status
+  if (gameState == "computer") {
+    playerRPS = giveRandomRPS();
+  }
+
+  // Check reverse mode status
+  if (checkReverse == 1) {
+    verifyWinner = didPlayerWin(botRPS, playerRPS);
+  } else {
+    verifyWinner = didPlayerWin(playerRPS, botRPS);
+  }
+
+  // Check Korean mode status
+  if (gameState == "korean") {
+    koreanWinner = koreanRPS(verifyWinner);
+  }
+
+  // End of game - Record and return results
+  scoreTracker(verifyWinner);
+
+  var myOutput = `
+  You input ${playerRPS}, the bot input ${botRPS}.</br> 
+  ${userName} ${verifyWinner}s! 
+  The current score is ${playerWins}W-${playerLosses}L-${playerDraws}D. </br>
+  ${playerFeedback(playerWins, playerLosses)}
+  ${koreanWinner}`;
+
+  lastWinner = verifyWinner;
+
+  return myOutput;
+};
+
+// Returns true on non-RPS responses
+var inputValidator = function (input) {
+  if (gameState == "computer") {
+    return false;
+  }
+  if (
+    input != "scissors" &&
+    input != "paper" &&
+    input != "stone" &&
+    input != "reversed scissors" &&
+    input != "reversed paper" &&
+    input != "reversed stone" &&
+    input != "reverse"
+  ) {
+    return true;
+  }
+};
+
+// Return 1 to reverse results if "reversed" is input
+var determineReverse = function (string) {
+  if (string.split(" ")[0] == "reversed") {
+    return 1;
+  } else {
+    return 0;
+  }
```

Comment:
you can just put return 0 instead of having an else since the return 1 in line 103 already negates the need for an else statement afterwards